sklearn/neighbors/_binary_tree.pxi.tp

```diff
@@ -806,9 +806,9 @@ cdef class BinaryTree{{name_suffix}}:
     cdef readonly const {{INPUT_DTYPE_t}}[::1] sample_weight
     cdef public float64_t sum_weight
 
-    cdef public const intp_t[::1] idx_array
+    cdef public intp_t[::1] idx_array
     cdef public const NodeData_t[::1] node_data
-    cdef public const {{INPUT_DTYPE_t}}[:, :, ::1] node_bounds
+    cdef public {{INPUT_DTYPE_t}}[:, :, ::1] node_bounds
```

Comment:
While I agree that those must be not be `const` and that `const`-ness seems inconsistent in Cython, this makes serialization fail.

Could you revert those changes for now?